.pre-commit-config.yaml

```diff
@@ -63,5 +63,5 @@ repos:
           - pytest
           - tornado
           - pyarrow
-          - git+https://github.com/dask/dask
+          - git+https://github.com/dask/dask.git
```

Comment:
Thanks for figuring out the pre-commit caching issue. I got so distracted by it that I forgot to streamline tests with dask/dask#10157.

Reply:
> Okay, so it appears there's some sort of caching happening in our pre-commit build (outside the caching we implement ourselves as part of the normal test suite runs). 

There is indeed. The [pre-commit action](https://github.com/pre-commit/action/blob/main/action.yml) does this:

https://github.com/pre-commit/action/blob/5f528da5c95691c4cf42ff76a4d10854b62cbb82/action.yml#L15-L20

So that hashes the contents of the `pre-commit-config.yaml` to determine whether or not to restore from a cache. But that is definitely unsafe for the `additional_dependencies` key here unless exact versions are pinned and then updated when moved forward (since, as we notice here, dask can change without the URL to the dask repo changing).

If the caching is more trouble than it's worth, one could probably just remove the use of the "official" pre-commit action given that all it does is "pip install pre-commit; pre-commit run ..."

Reply:
FWIW, one can manually delete caches at https://github.com/dask/distributed/actions/caches.